ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+
+```solidity
+interface IStandard {
+    /**
+     * Validate user's intent
+     *
+     * @dev returning validation result, the type uses bytes4 for extensibility purpose
+     * @return result values representing validation outcomes
+     */
+    function validateUserIntent(bytes calldata intent) external view returns (bytes4 result);
+
+    /**
+     * Unpack user's intent, it is RECOMMENDED to validate intent while unpacking to save gas
+     *
+     * @dev returning unpacked result, the type uses bytes for extensibility purpose
+     * @return result unpacked result status
+     * @return operations unpacked operations that can be executed by the IAccount, NOT REQUIRED to match UserIntent.instructions
+     */
+    function unpackOperations(bytes calldata intent) external view returns (bytes4 memory result, bytes[] memory operations);
+}
+```
+
+Notice that `IStandard` fully controls the validation process, similar to `EntryPoint` in ERC-4337 and ERC-7521,
+each `IStandard` implementation MUST be pre-audited and SHOULD NOT be upgradable.
+
+### IAccount interface
+
+On the account side, `IAccount` is the interface to execute `bytes calldata intent`:
+
+```solidity
+interface IAccount {
+    /**
+     * Execute user's intent
+     * 
+     * @dev returning execution result, the type uses bytes for extensibility purpose
+     * @return result values representing execution outcomes
+     */
+    function executeUserIntent(bytes calldata intent) external returns (bytes memory);
+}
+```
+
+Using `SET_CODE_TX_TYPE=0x04`, each account is an implementation of `IAccount` interface and EOAs can delegate the contract code to the account. Account contract MUST be audited because it controls access to EOAs' assets. Thanks to the design that different EOAs can delegate to the same account implementation, each account only need to be deployed and audited once. It has to be the EOA account owners'
+responsibility to delegate to a safe `IAccount` implementation.
+
+It is RECOMMENDED that each account leverages `IStandard` to validate and unpack operations, check **Reference
+Implementation** for examples. Account smart contract can be stateless to avoid sharing storage space with other delegated contracts.
+
+### Usage of Bytes
+
+There are many objects defined with `bytes` type for extensibility, future-compatibility purpose. All those objects are
+optional and their usages depends on `IStandard` and `IAccount` implementation.
+
+## Rationale
+
+> The main challenge with a generalized intent standard is being able to adapt to the evolving world of intents. Users
+> need to have a way to express their intents in a seamless way without having to make constant updates to their smart
+> contract wallets. --- ERC-7521
+
+The interface designs proposed in this EIP is inspired by ERC-7521, and the author tries to avoid turning EIP into a
+real implementation.
```

Comment:
You don't need to quote 7521 here. Just write your rationale describing the decisions made within this document. If you're going to point to another EIP in this section, that is a hint that you're not actually writing content appropriate for the Rational section. My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We painted the shed red because...

Reply:
Updated 👍